### ISSUE TITLE
Inherit a sub class from a super class failes when the super class has already imported the sub class

### ISSUE THREAD

**Issue by tobimx** [webpack/webpack#2554]
Hi, i'am running webpack 1.13.1

``` js
//works
import {SuperClass} from '../super_class';
class SubClass extends SuperClass () {}

//doesn't work
//TypeError: Super expression must either be null or a function, not undefined
import {SuperClass} from '../main';
class SubClass extends SuperClass () {}
```

**Comment by tobimx** [comment]
I'am digging more into it and I think this is somehow a cycle problem:

~~~js
/*
 * build.js
 *   - uses SuperClass
 *      - uses SubClass 
 */
~~~

Now I want to inherit SubClass from SuperClass and that seems to be the problem. In my opinion it should work but I'll redesign my implementation

**Comment by sokra** [comment]
It's a circular dependency.

**Comment by tobimx** [comment]
Yes somehow, but I also tried this:

{code}
class Abstract () {}
class Sub extends Abstract () {}

import Sub from './sub';
class Super extends Abstract () {
  constructor () {
    this.sub = new Sub();
  }
}
{code}

I think this should work but it doesn't.
Think of a composition tree where I want to add childs of the same abstract class but different implementation.

**Comment by sokra** [comment]
Do you export the class with `export default class Sub from ...`

**Comment by tobimx** [comment]
No, I'am using

~~~js
export {Class}
~~~

**Comment by pokorson** [comment]
Hi, I've found similiar issue in my application. It looks like circular dependency but not so obvious.
Would appreciate any suggestions how to deal with it.

I have directory
```
components/
    A.js   
    B.js
    index.js
```

in `index.js` I'm reexporting modules like this
```javascript
import A from './A';
import B from './B';

export {A, B};
```

And each file
```javascript
/* A.js */
export default class A extends React.Component {
     ...
}
```

```javascript
/* B.js */
import { A } from 'components';
export default class B extends A {
     ...
}
```
When I want to import module `A` from third file like this 
`import { A } from 'components'`

I'm getting `  TypeError: Super expression must either be null or a function, not undefined`
It looks to me it's due to loading `A` via `index.js` which loads also `B` which extends `A`.
And also I've created alias for components so all modules are resolved fine.
Should I avoid reexporting modules in such way?
